content/lessons/functional-programming/pure-functions.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Pure Functions
+template: lesson
+draft: false
+slug: /courses/Functional-Programming/pure-functions
+course: Functional-Programming
+tags:
+  - "Usability"
+  - "Maintainability"
+  - "Functional Programming"
+  - "Pure Functions"
+description: |
+  The second lesson of the Awesome Learning Functional Programming course will teach students how to apply the concept of functional purity to your code and how to write and recognize pure functions.
+videoLinks:
+  - https://www.youtube.com/embed/dZ41D6LDSBg
+  - https://www.youtube.com/embed/fYbhD_KMCOg
+readingLinks:
+  - link: https://www.freecodecamp.org/news/what-is-a-pure-function-in-javascript-acb887375dfe/
+    title: What Is a Pure Function in JavaScript?
+    description: 5 minute read. General introduction to pure functions in JavaScript.
+timeToCompletion: ~1 hour
+preReadQuizLink: 'https://docs.google.com/forms/d/e/1FAIpQLSeD57l2LvyDKM9Rhup0h0q6ltZbGJZi7lP56ETFAvBjugulZQ/viewform?usp=sf_link'
+secondaryExerciseUrl: 'https://codesandbox.io/s/github/wayfair/awesome-learning-exercises/tree/master/functional-programming/pure-functions'
```

Comment:
1. You shouldn't need this secondary URL; it should ideally just be automatic based on the directory of this lesson. Can you check if the CodeSandbox link still works without this link?
2. Branches for both this repo and the exercise repo should refer to `main` now!